pkg/registry/core/service/storage/rest.go

```diff
@@ -222,32 +225,35 @@ func (rs *REST) Delete(ctx context.Context, id string, options *metav1.DeleteOpt
 
 	svc := obj.(*api.Service)
 
-	// TODO: can leave dangling endpoints, and potentially return incorrect
-	// endpoints if a new service is created with the same name
-	_, _, err = rs.endpoints.Delete(ctx, id, &metav1.DeleteOptions{})
-	if err != nil && !errors.IsNotFound(err) {
-		return nil, false, err
-	}
-
-	if helper.IsServiceIPSet(svc) {
-		rs.serviceIPs.Release(net.ParseIP(svc.Spec.ClusterIP))
-	}
+	// Only perform the cleanup if this is a non-dryrun deletion
+	if !dryrun.IsDryRun(options.DryRun) {
```

Comment:
Would the code be more readable if the code in this `if` were moved into its own function? Up to you to decide if it's worth it.

Reply:
I think the most readable might be to leave the rs.endpoints.Delete call and error handling for that here, and move the rest of the cleanup to another function called something like "releaseAllocatedResources", to release the ports and ips